src/components/oInput.js

```diff
@@ -0,0 +1,227 @@
+  });
+  return this;
+}
+
+oInput.prototype.disabled = function(required) {
+  if(typeof disabled !== 'boolean') return this;
+
+  this.element.attr({
+    disabled
+  });
+  return this;
+}
+
+oInput.prototype.readOnly = function(readonly) {
+  if(typeof readonly !== 'boolean') return this;
+
+  this.element.attr({
+    readonly
+  });
+  return this;
+}
+
+oInput.prototype.id = function (id) {
+  if(typeof id === 'undefined') return this;//nie wiem czy sprawdzać też pusty string + inne walidacje
+
+  this.element.id(id);
+  return this;
+}
+
+oInput.prototype.onChange = function(event) {
+  if(typeof event !== 'function') return this;
+
+  this.element.event({
+    name: 'change',
+    fn: e => event(e, this)
+  });
+  return this;
+}
+
+oInput.prototype.onKeyUp = function(event) {
+  if(typeof event !== 'function') return this;
+
+  this.element.event({
+    name: 'keyup',
+    fn: e => event(e, this)
+  });
+  return this;
+}
+
+oInput.prototype.onKeyDown = function(event) {
+  if(typeof event !== 'function') return this;
+
+  this.element.event({
+    name: 'keydown',
+    fn: e => event(e, this)
+  });
+  return this;
+}
+
+oInput.prototype.onFocus = function(event) {
+  if(typeof event !== 'function') return this;
+
+  this.element.event({
+    name: 'focusin',
+    fn: e => event(e, this)
+  });
+  return this;
+}
+
+oInput.prototype.onFocusOut = function(event) {
+  if(typeof event !== 'function') return this;
+
+  this.element.event({
+    name: 'focusout',
+    fn: e => event(e, this)
+  });
+  return this;
+}
+
+oInput.prototype.value = function(value) {
+  this.element.setValue(value);
+  return this;
+}
+
+oInput.prototype.getValue = function() {
+  return this.input.value;
+}
+oInput.prototype.placeholder = function(placeholder) {
+  this.element.attr({
+    placeholder
+  });
+  return this;
+}
+
+oInput.prototype.classList = function(classList) {
+  this.element.classList(classList);
+  return this;
+}
+
+oInput.prototype.index = function(index) {
+  this.index = index;
+  return this;
+}
+
+oInput.prototype.init = function() {
+  return this.element.init();
+}
+
+//TODO: 2) e.target.value of type number
+oInput.prototype.db = function (db, name, updateOn = 'change') {
+  if(!isValidInputEventName(updateOn)) {
+    updateOn = 'change';
+  }
+
+  if(this.dbSet) return this;
+
+  this.dbSet = true;
+  this.dbObject = db;
+  this.dbKey = name;
+
+  this.element.event({
+    name: updateOn,
+    fn: e => {
+      updateComponentDb(
+        this.dbObject,
+        this.dbKey,
+        this.index,
+        e.target.value
+      );
+    }
+  });
+
+  this.input.value = this.index ? db[this.index][name] : db[name];
+  return this;
+}
+
+//TODO: newValue validation (?)
+oInput.prototype.formatter = function(formatterFunction, formatOn = 'keyup') {
+  if(typeof formatterFunction !== 'function') return this;
+  if(!isValidInputEventName(formatOn)) {
+    formatOn = 'keyup';
+  }
+
+  this.element.event({
+    name: formatOn,
```

Comment:
Co to za event formatOn? Jak to działa? te this.element.event to jest z ojs-core?

Reply:
Chodzi o to na jaki event ustawić tę metodę. Jak podasz change, to funkcja się wykona na onChange itd. Tak this.element.event jest z ojs-core

Reply:
To moja ślepota. Wydawało mi się, że formatOn jest w stringu, Udajmy, że nie było pytania